tests/test_ops.py

```diff
@@ -0,0 +1,228 @@
+    a = test_df_year.filter(scenario='scen_a', variable='Primary Energy')
+    b = test_df_year.filter(scenario='scen_a', variable='Primary Energy|Coal')
+    
+    op = BinaryOp(a, b)
+    axis = 'variable'
+    axis_value = 'Non-Coal PE'
+    obs_data, obs_meta = op.calc(subtract_op, axis=axis, axis_value=axis_value)
+
+    # test data
+    exp = a.copy()
+    exp['value'] = [0.5, 3.0]
+    exp['variable'] = [axis_value, axis_value]
+    idx = LONG_IDX + ['value']
+    assert_frame_equal(obs_data[idx], exp.data[idx])
+
+    # test meta
+    assert_frame_equal(obs_meta, exp.meta)
+
+def test_calc_subtract_self(test_df_year):
+    a = test_df_year.filter(scenario='scen_a', variable='Primary Energy')
+    
+    op = BinaryOp(a, a)
+    axis = 'variable'
+    axis_value = 'Non-Coal PE'
+    obs_data, obs_meta = op.calc(subtract_op, axis=axis, axis_value=axis_value)
+
+    # test data
+    exp = a.copy()
+    exp['value'] = [0.0, 0.0]
+    exp['variable'] = [axis_value, axis_value]
+    idx = LONG_IDX + ['value']
+    assert_frame_equal(obs_data[idx], exp.data[idx])
+
+    # test meta
+    assert obs_meta.empty
+    assert_frame_equal(a.meta, exp.meta)
+    
+    
+def test_calc_subtract_axis_scenario(test_df_year):
+    a = test_df_year.filter(scenario='scen_a', variable='Primary Energy')
+    b = test_df_year.filter(scenario='scen_b', variable='Primary Energy')
+    
+    op = BinaryOp(a, b)
+    axis = 'scenario'
+    axis_value = 'scen_a - scen_b'
+    obs_data, obs_meta = op.calc(subtract_op, axis=axis, axis_value=axis_value)
+
+    # no meta test needed as meta is being messed with here (scenarios)
+    exp = a.copy()
+    exp['value'] = [-1.0, -1.0]
+    exp['scenario'] = [axis_value, axis_value]
+    idx = LONG_IDX + ['value']
+    assert_frame_equal(obs_data[idx], exp.data[idx])
+
+def test_calc_subtract_iamdataframe(test_df_year):
+    a = test_df_year.filter(scenario='scen_a', variable='Primary Energy')
+    b = test_df_year.filter(scenario='scen_a', variable='Primary Energy|Coal')
+
+    axis = 'variable'
+    axis_value = 'Non-Coal PE'
+    obs = a.subtract(b, axis=axis, new_name=axis_value)
+    
+    # test data
+    exp = a.copy()
+    exp['value'] = [0.5, 3.0]
+    exp['variable'] = [axis_value, axis_value]
+    idx = LONG_IDX + ['value']
+    assert_frame_equal(obs.data[idx], exp.data[idx])
+
+    # test meta
+    assert_frame_equal(obs.meta, exp.meta)
+
+def test_calc_subtract_inplace(test_df_year):
+    df = test_df_year.filter(scenario='scen_a')
+    a = 'Primary Energy'
+    b = 'Primary Energy|Coal'
+    axis = 'variable'
+    axis_value = 'Non-Coal PE'
+
+    # exp value
+    exp = df.filter(variable='Primary Energy').copy()
+    exp['value'] = [0.5, 3.0]
+    exp['variable'] = [axis_value, axis_value]
+
+    # test append=False
+    idx = LONG_IDX + ['value']
+    obs = df.subtract_inplace(a, b, axis=axis, new_name=axis_value,
+                              append=False)
+    assert_frame_equal(obs.data[idx], exp.data[idx])
```

Comment:
as above, should be possible to use `pyam.testing.assert_iamframe_equal()` here

Reply:
done